experimental/traffic-portal/src/app/core/types/detail/type-detail.component.html

```diff
@@ -0,0 +1,49 @@
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<mat-card>
+	<tp-loading *ngIf="!type"></tp-loading>
+	<form ngNativeValidate (ngSubmit)="submit($event)" *ngIf="type">
+		<mat-card-content>
+			<mat-form-field *ngIf="!new">
+				<mat-label>ID</mat-label>
+				<input matInput type="text" name="id" disabled readonly [defaultValue]="type.id" />
+			</mat-form-field>
+			<mat-form-field>
+				<mat-label>Name</mat-label>
+				<input matInput type="text" name="name" required [(ngModel)]="type.name" />
+			</mat-form-field>
+			<mat-form-field>
+				<mat-label>Description</mat-label>
+				<input matInput type="text" name="description" required [(ngModel)]="type.description" />
+			</mat-form-field>
+			<mat-form-field *ngIf="!new">
+				<mat-label>Use In Table</mat-label>
+				<input matInput type="text" name="useInTable" disabled readonly [(ngModel)]="type.useInTable" />
```

Comment:
2-way binding is unnecessary when the control is read-only; this should probably use `defaultValue` just like the ID field.